From bfd30b3dac42ec1aa96b80ccc75dc1b668f7ee04 Mon Sep 17 00:00:00 2001
From: R0rt1z2 <me@r0rt1z2.com>
Date: Mon, 9 Oct 2023 10:24:46 +0200
Subject: [PATCH] libshim_showlogo: Port to U

* Updated mangled symbol names for Android 14 and addressed ABI issues related to improper
  return types and handling of the class pointer ('this').

* In createSurface, pass actual parameters instead of null pointers by obtaining the handle
  of the SurfaceControl parent, as referenced in commit 992496bd [1].

* Based on commit a7fe3048 [2], the switch to DisplayMode was primarily a rename; removed
  unnecessary casting and the associated header file.

* Also format the shim with `clang-format`.

- [1]: https://github.com/LineageOS/android_frameworks_native/commit/992496bd2e1af9c93bce33216034dd5419acb444#diff-b8bfdcf33cdb1b09a6a8f0438d5d75cb33d25cba14b276846d8b8ecbd10a70deR1631
- [2]: https://github.com/LineageOS/android_frameworks_native/commit/a7fe3048aaaf8a45337c7355326f2c6b68b70fb0
---
 libshims/Android.bp              |  1 -
 libshims/include/DisplayConfig.h | 39 --------------------------
 libshims/libshim_showlogo.cpp    | 52 +++++++++++++++++++++++------------
 3 files changed, 34 insertions(+), 58 deletions(-)
 delete mode 100644 shims/include/DisplayConfig.h

diff --git a/libshims/Android.bp b/libshims/Android.bp
index a82aebe..766f913 100644
--- a/libshims/Android.bp
+++ b/libshims/Android.bp
@@ -51,5 +51,4 @@ cc_library_shared {
         "libgui",
         "libutils",
     ],
-    local_include_dirs: ["include"]
 }
diff --git a/libshims/include/DisplayConfig.h b/libshims/include/DisplayConfig.h
deleted file mode 100644
index 3cd2687..0000000
--- a/libshims/include/DisplayConfig.h
+++ /dev/null
@@ -1,39 +0,0 @@
-/*
- * Copyright 2019 The Android Open Source Project
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-#pragma once
-
-#include <type_traits>
-
-#include <ui/Size.h>
-#include <utils/Timers.h>
-
-namespace android {
-
-// Configuration supported by physical display.
-struct DisplayConfig {
-    ui::Size resolution;
-    float xDpi = 0;
-    float yDpi = 0;
-
-    float refreshRate = 0;
-    nsecs_t appVsyncOffset = 0;
-    nsecs_t sfVsyncOffset = 0;
-    nsecs_t presentationDeadline = 0;
-    int configGroup = -1;
-};
-
-} // namespace android
diff --git a/shims/libshim_showlogo.cpp b/shims/libshim_showlogo.cpp
index c816d5a..d151513 100644
--- a/libshims/libshim_showlogo.cpp
+++ b/libshims/libshim_showlogo.cpp
@@ -1,35 +1,51 @@
-#include <DisplayConfig.h>
-#include <ui/Size.h>
 #include <gui/SurfaceComposerClient.h>
+#include <ui/Size.h>
 
 using namespace android;
 
 extern "C" {
 
-void _ZN7android21SurfaceComposerClient13createSurfaceERKNS_7String8EjjijRKNS_2spINS_7IBinderEEENS_13LayerMetadataEPj(const String8& name, uint32_t w, uint32_t h, PixelFormat format, uint32_t flags, const sp<IBinder>& parentHandle, LayerMetadata metadata, uint32_t* outTransformHint);
-
-void _ZN7android21SurfaceComposerClient13createSurfaceERKNS_7String8EjjijPNS_14SurfaceControlENS_13LayerMetadataEPj(const String8& name, uint32_t w, uint32_t h, PixelFormat format, uint32_t flags, SurfaceControl* parent, LayerMetadata metadata, uint32_t* outTransformHint) {
-  _ZN7android21SurfaceComposerClient13createSurfaceERKNS_7String8EjjijRKNS_2spINS_7IBinderEEENS_13LayerMetadataEPj(name, w, h, format, flags, nullptr, metadata, nullptr);
- }
+void *
+_ZN7android21SurfaceComposerClient13createSurfaceERKNS_7String8EjjiiRKNS_2spINS_7IBinderEEENS_3gui13LayerMetadataEPj(
+    void *thisptr, const String8 &name, uint32_t w, uint32_t h,
+    PixelFormat format, uint32_t flags, const sp<IBinder> &parentHandle,
+    LayerMetadata metadata, uint32_t *outTransformHint);
+
+void *
+_ZN7android21SurfaceComposerClient13createSurfaceERKNS_7String8EjjijPNS_14SurfaceControlENS_13LayerMetadataEPj(
+    void *thisptr, const String8 &name, uint32_t w, uint32_t h,
+    PixelFormat format, uint32_t flags, SurfaceControl *parent,
+    LayerMetadata metadata, uint32_t *outTransformHint) {
+  return _ZN7android21SurfaceComposerClient13createSurfaceERKNS_7String8EjjiiRKNS_2spINS_7IBinderEEENS_3gui13LayerMetadataEPj(
+      thisptr, name, w, h, format, flags, parent->getHandle(), metadata, outTransformHint);
+}
 
-void _ZN7android14SurfaceControl10getSurfaceEv(void);
+void *_ZN7android14SurfaceControl10getSurfaceEv(void *thisptr);
 
-void _ZNK7android14SurfaceControl10getSurfaceEv(void) {
-  _ZN7android14SurfaceControl10getSurfaceEv();
- }
+void *_ZNK7android14SurfaceControl10getSurfaceEv(void *thisptr) {
+  return _ZN7android14SurfaceControl10getSurfaceEv(thisptr);
+}
 
-status_t _ZN7android21SurfaceComposerClient20getActiveDisplayModeERKNS_2spINS_7IBinderEEEPNS_2ui11DisplayModeE(const sp<IBinder>& display, ui::DisplayMode* mode);
+status_t
+_ZN7android21SurfaceComposerClient20getActiveDisplayModeERKNS_2spINS_7IBinderEEEPNS_2ui11DisplayModeE(
+    void *thisptr, const sp<IBinder> &display, ui::DisplayMode *mode);
 
-status_t _ZN7android21SurfaceComposerClient22getActiveDisplayConfigERKNS_2spINS_7IBinderEEEPNS_13DisplayConfigE(const sp<IBinder>& display, DisplayConfig* config) {
-  return _ZN7android21SurfaceComposerClient20getActiveDisplayModeERKNS_2spINS_7IBinderEEEPNS_2ui11DisplayModeE(display, (ui::DisplayMode*)config);
+status_t
+_ZN7android21SurfaceComposerClient22getActiveDisplayConfigERKNS_2spINS_7IBinderEEEPNS_13DisplayConfigE(
+    void *thisptr, const sp<IBinder> &display, ui::DisplayMode *config) {
+  return _ZN7android21SurfaceComposerClient20getActiveDisplayModeERKNS_2spINS_7IBinderEEEPNS_2ui11DisplayModeE(
+      thisptr, display, config);
 }
 
-status_t _ZN7android21SurfaceComposerClient11Transaction5applyEbb(void* thisptr, bool synchronous, bool oneWay);
+status_t _ZN7android21SurfaceComposerClient11Transaction5applyEbb(
+    void *thisptr, bool synchronous, bool oneWay);
 
-status_t _ZN7android21SurfaceComposerClient11Transaction5applyEb(void* thisptr, bool synchronous) {
-    return _ZN7android21SurfaceComposerClient11Transaction5applyEbb(thisptr, synchronous, false /* oneWay */);
+status_t
+_ZN7android21SurfaceComposerClient11Transaction5applyEb(void *thisptr,
+                                                        bool synchronous) {
+  return _ZN7android21SurfaceComposerClient11Transaction5applyEbb(
+      thisptr, synchronous, false /* oneWay */);
 }
 
 ui::Size _ZN7android2ui4Size7INVALIDE;
-
 }
-- 
2.30.2

